app/models/ability.rb

```diff
@@ -0,0 +1,39 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the passed in user here. For example:
+    #
+    #   user ||= User.new # guest user (not logged in)
+    #   if user.admin?
+    #     can :manage, :all
+    #   else
+    #     can :read, :all
+    #   end
+    #
+    # The first argument to `can` is the action you are giving the user
+    # permission to do.
+    # If you pass :manage it will apply to every action. Other common actions
+    # here are :read, :create, :update and :destroy.
+    #
+    # The second argument is the resource the user can perform the action on.
+    # If you pass :all it will apply to every resource. Otherwise pass a Ruby
+    # class of the resource.
+    #
+    # The third argument is an optional hash of conditions to further filter the
+    # objects.
+    # For example, here the user can only update published articles.
+    #
+    #   can :update, Article, :published => true
+    #
+    # See the wiki for details:
+    # https://github.com/CanCanCommunity/cancancan/wiki/Defining-Abilities
+
+    can :destroy, Post, author: user
+    can :destroy, Comment, author: user
+    return unless user.role == 'admin' # additional permissions for administrators
+
+    can :destroy, Post
+    can :destroy, Comment
```

Comment:
- [ ] Owner of the post not only destroy, but they can perform all view, update or destroy own post or comment.

- [ ] And the admin role can manage all.

 You can write like this:
```ruby

user ||= User.new # guest user (not logged in)

if user.admin? 
 can :manage, :all
else
 can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
 can :manage, Comment, author: user
 can :read, :all
end

```